### PR TITLE
fix completely ignored dependencies querying for updates

### LINKED ISSUE #7651: Ignore doesn't ignore
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Package ecosystem

Docker

### dependabot.yml content

  - package-ecosystem: "docker"
    directory: "/"
    schedule:
      interval: "daily"
    ignore:
      - dependency-name: "xyz/*"

### What you expected to see, versus what you actually saw

At the moment the dependencies are still checked and appear in the logs as
```
updater | 2023/07/19 09:16:17 INFO <job_697229887> Checking if xyz latest needs updating
updater | 2023/07/19 09:16:17 INFO <job_697229887> Ignored versions:
updater | 2023/07/19 09:16:17 INFO <job_697229887>   >= 0 - from .github/dependabot.yml
updater | 2023/07/19 09:16:17 INFO <job_697229887> Latest version is latest
 proxy | 2023/07/19 09:16:17 [020] HEAD xyz
```
As the check might fail intentionally (let's say because of a dependency in a private repository), the dependency would still fail the dependabot task (and count for the 30 trials until dependabot refuses to work).

I would expect them to be ignored without a check and error.

### PR DESCRIPTION
**Pull request by jakecoffman**
Fixes #7651

If you try to ignore all versions of a dependency that is otherwise allowed to update, Dependabot will still query the registry to get a list of versions.

This becomes a problem if the dependency source is an un-configured private registry because it will fail the HTTP request and cause an error on the job which is completely avoidable if Dependabot had skipped the check.

This PR adds a check for completely ignored dependencies in `allowed_update?` which will fix the issue for individual PRs and grouped updates. 

Future enhancements to this could include checking if the range provided means no update is possible, like if the dependency is at `1.0.0` and the ignored version range is `> 1.0.0`.

### CODE REVIEW

File: updater/lib/dependabot/job.rb [hunk 1]
```diff
@@ -171,7 +171,14 @@ def reject_external_code?
     # separately, if required.
     #
     # rubocop:disable Metrics/PerceivedComplexity
+    # rubocop:disable Metrics/CyclomaticComplexity
     def allowed_update?(dependency)
+      # Ignoring all versions is another way to say no updates allowed
+      if completely_ignored?(dependency) && !security_updates_only?
```
Comment (abdulapopoola) [inline updater/lib/dependabot/job.rb:177]: should this condition be in the `completely_ignored` method? Also, for my curiosity, any reason why security_updates have to be false?

Finally; any correlation between this and the ongoing alert rules by @landongrindheim and @bdragon ?

**Review by abdulapopoola — APPROVED** [review]
approving with comments

File: updater/lib/dependabot/job.rb (hunk 1 above)
Comment (deivid-rodriguez) [reply updater/lib/dependabot/job.rb:177]: I had the same thought as @abdulapopoola, but also `ignore_conditions_for` method is already passing around the `security_updates_only?` flag, so maybe whatever the correct behavior for security updates is, it is already being considered?

Comment (jakecoffman) [reply updater/lib/dependabot/job.rb:177]: Ah, didn't notice it was passed in `ignore_conditions_for` already, thanks @deivid-rodriguez I'll remove it.

Comment (jakecoffman) [reply updater/lib/dependabot/job.rb:177]: For alert rules, we shouldn't even get in the job, it will ignore before creating the job itself.

**Pull request merged**